### PR TITLE
[ESI-14308] Added traceID to golang extensions and modified environment copying

### PR DESCRIPTION
**Pull request by andriikonstantinov**

### CODE REVIEW

**Comment by zimnx** [comment]
I don't like that we spoil interface with parameter which is used only for debugging. TraceID should be either tied with context or logger itself. I asked you to look into GZTP code, you can find there these two strategies. Please consult your next proposal with someone before you start working on it.

File: cli/cli.go [hunk 1]
```diff
@@ -468,15 +470,15 @@ func getMigrateSubcommandWithPostMigrateEvent(subcmd, usage string) cli.Command
...
 				return
 			}
 
-			log.Debug("Modified schemas: %s", modifiedSchemas)
+			log.Debug("[%s] Modified schemas: %s", traceID, modifiedSchemas)
 
 			schemaFiles := config.GetStringList("schemas", nil)
 			if schemaFiles == nil {
-				log.Fatal("No schema specified in configuraion")
+				log.Fatal("[%s] No schema specified in configuraion", traceID)
```
Comment (p-kozlowski) [inline cli/cli.go:481]: should be Fatalf (same elsewhere, no fmt.Sprintf needed)

File: server/snmp.go [hunk 2]
```diff
@@ -23,11 +23,13 @@ import (
 
 	l "github.com/cloudwan/gohan/log"
 	"github.com/cloudwan/gohan/util"
+	"github.com/twinj/uuid"
 )
 
 //SNMP Process
 //Experimental
 func startSNMPProcess(server *Server) {
+	traceID := uuid.NewV4().String()
```
Comment (p-kozlowski) [inline server/snmp.go:32]: move generating traceId inside the for loop

File: extension/goplugin/environment.go [hunk 3]
```diff
@@ -144,6 +146,14 @@ func (thisEnvironment *Environment) Database() goext.IDatabase {
 	return thisEnvironment.extDatabase
 }
 
+func (thisEnvironment *Environment) SetEnvBindings() {
```
Comment (p-kozlowski) [inline extension/goplugin/environment.go:149]: this function is not Setting, it's Reseting. I'd go for "ResetEnvBindings", or simply "Rebind"

**Pull request closed without merging**